cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CaReconciler.java

```diff
@@ -373,6 +385,64 @@ Future<Void> rollingUpdateForNewCaKey() {
         }
     }
 
+    /**
+     * Gather the Kafka related components pods for checking CA key trust and old CA certificates during the reconcile
+     * It's done one time at the beginning of the reconcile to avoid an expensive operation like listing pods twice
+     *
+     * @return the list of Kafka related components pods
+     */
+    Future<Void> listPods() {
+        // Building the selector for Kafka related components
+        Labels labels =  Labels.forStrimziCluster(reconciliation.name()).withStrimziKind(Kafka.RESOURCE_KIND);
+
+        return podOperator.listAsync(reconciliation.namespace(), labels)
+                .compose(pods -> {
+                    this.pods.clear();
+                    if (pods.isEmpty()) {
+                        return Future.succeededFuture();
+                    }
+                    for (Pod pod : pods) {
+                        // with "RollingUpdate" strategy on Deployment(s) (i.e. the Cruise Control one),
+                        // while the Deployment is reported to be ready, the old pod is still alive but terminating
+                        // this condition is for skipping "Terminating" pods for checks on the CA key and old certificates
+                        if (pod.getMetadata().getDeletionTimestamp() == null) {
+                            this.pods.add(pod);
```

Comment:
Wouldn't it make sense to instead of storing all the pods in the object to run the two checks on them and store the booleans instead? It seems like that would safe a lot of memory.